fastlane_core/lib/fastlane_core/print_table.rb

```diff
@@ -34,16 +41,60 @@ def print_values(config: nil, title: nil, hide_keys: [], mask_keys: [])
         return params
       end
 
-      def limit_row_size(rows, max_length = 100)
+      def colorize_array(array, colors)
+        value = ""
+        array.each do  |l|
+          colored_line = l
+          colored_line = "#{colors.first[0]}#{l}#{colors.last[0]}" if colors.length > 0
+          value << colored_line
+          value << "\n"
+        end
+        return value
+      end
+
+      def transform_output(rows, transform)
         require 'fastlane_core/string_filters'
+        require 'io/console'
+
+        return_array = []
+        tcols = IO.console.winsize[1]
+
+        col_count = rows.map(&:length).first || 1
+
+        # -4 per column - as tt adds "| " and " |"
+        terminal_table_padding = 4
+        max_length = tcols - (col_count * terminal_table_padding)
+
+        max_value_length = (max_length / col_count)
 
-        max_key_length = rows.map { |e| e[0].length }.max || 0
-        max_allowed_value_length = max_length - max_key_length - 7
         rows.map do |e|
-          value = e[1]
-          value = value.to_s.truncate(max_allowed_value_length) unless [true, false].include?(value)
-          [e[0], value]
+          new_row = []
+          e.each do |col|
+            value = col.to_s.dup
+            if transform == :truncate_middle
```

Comment:
would be nice to break this out into a transform_row/column function, because then appending to the row will be in a single place and be easier to maintain and test

i.e
```
def transform_row(transform, row)
  row.map do |column|
    transform_column(transform, column)
  end
end
```

```
def transform_column(transform, column)
  if transform == :truncate_middle
    ...
    value
  elseif transform == :newline
    ...
    value
end
```